src/Microsoft.DotNet.Build.Tasks/GenerateBindingRedirect.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using Microsoft.Build.Framework;
+using Task = Microsoft.Build.Utilities.Task;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class GenerateBindingRedirect : Task
+    {
+        [Required]
+        public ITaskItem[] AssemblyIdentities { get; set; }
+
+        [Required]
+        public ITaskItem[] Executables { get; set; }
+
+        [Required]
+        public string OutputPath { get; set; }
+
+        private static XNamespace ns { get; set; }
+
+        public override bool Execute()
+        {
+            ns = "urn:schemas-microsoft-com:asm.v1";
+            XElement bindingRedirectAssemblies = new XElement(ns + "assemblyBinding");
+            List<XElement> dependentAssemblyList = new List<XElement>();
+            foreach (ITaskItem assembly in AssemblyIdentities)
+            {
+                string publicKeyToken = assembly.GetMetadata("PublicKeyToken");
+                if (!string.IsNullOrEmpty(publicKeyToken))
```

Comment:
I typically try to avoid indentation when not needed.

You would write this as:
```csharp
if (string.IsNullOrEmpty(publicKeyToken))
{
    // maybe log a warning
    continue;
}

// the rest of the code.
```